### PR TITLE
Method to set PVC VolumeMode

### PR DESCRIPTION
**Pull request by yprokule**
Added method that defines what type of volume is required by the claim.

### CODE REVIEW

File: pkg/storage/pvc.go [hunk 1]
```diff
@@ -315,3 +344,17 @@ func (builder *PVCBuilder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// validateVolumeMode validates if requested volume mode is valid for PVC.
+func validateVolumeMode(volumeMode string) bool {
+	glog.V(100).Info("Validating volumeMode %s", volumeMode)
+
+	var validVolumeModes = map[string]string{
```
Comment (achuzhoy) [inline pkg/storage/pvc.go:352]: do we need to create a dict for 2 possible string values? i.e. why not just comparing to one of the strings?

File: pkg/storage/pvc.go [hunk 2]
```diff
@@ -150,6 +150,35 @@ func (builder *PVCBuilder) WithStorageClass(storageClass string) (*PVCBuilder, e
...
+		builder.errorMsg = fmt.Sprintf("Empty volumeMode requested for the PersistentVolumeClaim %s",
+			builder.Definition.Name)
+
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	if !validateVolumeMode(volumeMode) {
+		glog.V(100).Infof(fmt.Sprintf("Unsupported VolumeMode: %s", volumeMode))
+
+		builder.errorMsg = fmt.Sprintf("Unsupoorted VolumeMode %q requested for %s PersistentVolumeClaim",
```
Comment (achuzhoy) [inline pkg/storage/pvc.go:169]: misspell: Unsupoorted
Add the namespace

File: pkg/storage/pvc.go [hunk 3]
```diff
@@ -150,6 +150,35 @@ func (builder *PVCBuilder) WithStorageClass(storageClass string) (*PVCBuilder, e
...
+			volumeMode, builder.Definition.Name)
+
+		return builder, fmt.Errorf(builder.errorMsg)
+	}
+
+	nVolMode := v1.PersistentVolumeMode(volumeMode)
```
Comment (achuzhoy) [inline pkg/storage/pvc.go:175]: what is n is nVolMode ?

File: pkg/storage/pvc.go (hunk 1 above)
Comment (trewest) [reply pkg/storage/pvc.go:352]: Or just use the slices.Contains() function. Example - https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/rbac/clusterrolebinding.go#L85

File: pkg/storage/pvc.go (hunk 3 above)
Comment (yprokule) [reply pkg/storage/pvc.go:175]: normalized 😸 `builder.Definition.Spec.VolumeMode` is of type `*v1.PersistentVolumeMode` so U cannot just do conversion on the fly but have to use intermediary variable

File: pkg/storage/pvc.go (hunk 2 above)
Comment (yprokule) [reply pkg/storage/pvc.go:169]: thanks, updated

File: pkg/storage/pvc.go [hunk 4]
```diff
@@ -150,6 +150,35 @@ func (builder *PVCBuilder) WithStorageClass(storageClass string) (*PVCBuilder, e
...
 }
 
+// WithVolumeMode configures what type of volume is required by the claim.
+func (builder *PVCBuilder) WithVolumeMode(volumeMode string) (*PVCBuilder, error) {
+	glog.V(100).Infof("Set VolumeMode %s for the PersistentVolumeClaim", volumeMode)
+
+	if volumeMode == "" {
+		glog.V(100).Infof("Empty volumeMode requested for the PersistentVolumeClaim")
+
+		builder.errorMsg = fmt.Sprintf("Empty volumeMode requested for the PersistentVolumeClaim %s",
```
[the remark this answers is not in the document]
Comment (yprokule) [reply pkg/storage/pvc.go:160]: sure, added

File: pkg/storage/pvc.go (hunk 1 above)
Comment (yprokule) [reply pkg/storage/pvc.go:352]: thanks, updated

File: pkg/storage/pvc.go (hunk 3 above)
Comment (achuzhoy) [reply pkg/storage/pvc.go:178]: imho would be more beneficial to have a meaningful name,but up to you.

**Review by trewest — APPROVED** [review]
/lgtm - pending updates for Sasha's comment

File: pkg/storage/pvc.go (hunk 3 above)
Comment (yprokule) [reply pkg/storage/pvc.go]: updated and added comment

**Review by achuzhoy — APPROVED** [review]

**Review by cdvultur — APPROVED** [review]
/lgtm

**Pull request merged**